src/spring/azext_spring/_app_validator.py

```diff
@@ -138,3 +141,18 @@ def _get_app_name_from_namespace(namespace):
     elif hasattr(namespace, 'name'):
         return namespace.name
     return None
+
+
+def _validate_container_registry(cmd, namespace):
+    client = cf_spring(cmd.cli_ctx)
+    try:
+        build_service = client.build_service.get_build_service(namespace.resource_group,
+                                                               namespace.service,
+                                                               DEFAULT_BUILD_SERVICE_NAME)
+        if build_service.properties.container_registry:
+            if namespace.source_path or namespace.artifact_path:
+                raise InvalidArgumentValueError(
+                    "The instance using your own container registry can only use '--container-image' to deploy."
+                    " See more details in https://learn.microsoft.com/en-us/azure/spring-apps/how-to-deploy-with-custom-container-image?tabs=azure-cli")
+    except HttpResponseError:
+        pass
```

Comment:
Is it necessary to add some operations or error messages here?

Reply:
No, because if the build_service doesn't exist, we have another way to deploy apps. We only check the build service with user's own container registry here.